ambry-router/src/main/java/com.github.ambry.router/CloudRouterFactory.java

```diff
@@ -168,10 +174,35 @@ static RequestHandlerPool getRequestHandlerPool(VerifiableProperties verifiableP
     StoreKeyConverterFactory storeKeyConverterFactory =
         Utils.getObj(serverConfig.serverStoreKeyConverterFactory, verifiableProperties, registry);
     AmbryRequests requests =
-        new AmbryRequests(cloudStorageManager, channel, clusterMap, nodeId, registry, serverMetrics, null,
-            notificationSystem, null, storeKeyFactory, serverConfig.serverEnableStoreDataPrefetch,
-            storeKeyConverterFactory);
+        new AmbryRequests(cloudStorageManager, channel, clusterMap, nodeId, registry, serverMetrics, null, null, null,
+            storeKeyFactory, serverConfig.serverEnableStoreDataPrefetch, storeKeyConverterFactory);
     return new RequestHandlerPool(serverConfig.serverRequestHandlerNumOfThreads, channel, requests);
   }
+
+  private CompositeNetworkClientFactory getNetworkClientFactory(RequestHandlerPool requestHandlerPool,
+      MetricRegistry registry) {
+    NetworkConfig networkConfig = new NetworkConfig(verifiableProperties);
+    NetworkMetrics networkMetrics = new NetworkMetrics(registry);
+    NetworkClientFactory cloudNetworkClientFactory =
+        new LocalNetworkClientFactory((LocalRequestResponseChannel) requestHandlerPool.getChannel(), networkConfig,
+            networkMetrics, time);
+
+    NetworkClientFactory diskNetworkClientFactory;
+    if (new ClusterMapConfig(verifiableProperties).clusterMapHttp2NetworkClientEnabled) {
+      Http2ClientConfig http2ClientConfig = new Http2ClientConfig(verifiableProperties);
+      Http2ClientMetrics http2ClientMetrics = new Http2ClientMetrics(registry);
+      diskNetworkClientFactory = new Http2NetworkClientFactory(http2ClientMetrics, http2ClientConfig, sslFactory, time);
+    } else {
+      diskNetworkClientFactory = new SocketNetworkClientFactory(networkMetrics, networkConfig, sslFactory,
+          routerConfig.routerScalingUnitMaxConnectionsPerPortPlainText,
+          routerConfig.routerScalingUnitMaxConnectionsPerPortSsl, routerConfig.routerConnectionCheckoutTimeoutMs, time);
+    }
```

Comment:
Looks like this logic was copied from NonBlockingRouterFactory.  I'd prefer to see it reused somehow, though that can be done in a follow up.

Reply:
Made a static method in NonBlockingRouterFactory for this.